pkg/matcher/annotation_matcher.go

```diff
@@ -226,7 +236,19 @@ func MatchPipelinerunByAnnotation(ctx context.Context, logger *zap.SugaredLogger
 		if event.EventType == opscomments.NoOpsCommentEventType.String() || event.EventType == opscomments.OnCommentEventType.String() {
 			continue
 		}
+
 		if celExpr, ok := prun.GetObjectMeta().GetAnnotations()[keys.OnCelExpression]; ok {
+			// Checks if the Pipelinerun has `on-event`/`on-target-branch annotations` with `on-cel-expression`
+			// and if present then warns the user that `on-cel-expression` will take precedence
+			annotations := []string{}
+			annotations = append(annotations, appendAnnotationIfNotEmpty("on-event", prun.GetObjectMeta().GetAnnotations()[keys.OnEvent])...)
+			annotations = append(annotations, appendAnnotationIfNotEmpty("on-target-branch", prun.GetObjectMeta().GetAnnotations()[keys.OnTargetBranch])...)
+
+			ignoredAnnotations := strings.Join(annotations, ", ")
+			msg := fmt.Sprintf("Warning: The Pipelinerun '%s' has 'on-cel-expression' defined along with [%s] annotation(s). The `on-cel-expression` will take precedence and these annotations will be ignored", prun.Name, ignoredAnnotations)
+			logger.Warnf(msg)
+			eventEmitter.EmitMessage(repo, zap.WarnLevel, "RespositoryTakesOnCelExpressionPrecedence", msg)
```

Comment:
@PuneetPunamiya sorry forgot to suggest before...

```suggestion
			if len(annotations) > 0 {
		ignoredAnnotations := strings.Join(annotations, ", ")
		msg := fmt.Sprintf("Warning: The Pipelinerun '%s' has 'on-cel-expression' defined along with [%s] annotation(s). The `on-cel-expression` will take precedence and these annotations will be ignored", prun.Name, ignoredAnnotations)
		logger.Warnf(msg)
		eventEmitter.EmitMessage(repo, zap.WarnLevel, "RespositoryTakesOnCelExpressionPrecedence", msg)
	}
```